server/src/main/java/org/elasticsearch/action/search/TransportSearchShardsAction.java

```diff
@@ -86,7 +86,7 @@ public TransportSearchShardsAction(
     }
 
     @Override
-    protected void doExecute(Task task, SearchShardsRequest searchShardsRequest, ActionListener<SearchShardsResponse> listener) {
+    public void doExecute(Task task, SearchShardsRequest searchShardsRequest, ActionListener<SearchShardsResponse> listener) {
```

Comment:
We should instead have a separate method (like in field-caps) and make it very clear that this does _not_ run authz and that the caller is responsible for authorization.